.changeset/sharp-birds-rest.md

```diff
@@ -0,0 +1,16 @@
+---
+"@pnpm/plugin-commands-config": major
+"pnpm": major
+---
+
+The naming cases of top-level config keys have been changed for `pnpm config list [--json]` and `pnpm config get [--json]` (without argument).
+The specifics depend on the appearance of `--json` and the classification of the top-level keys.
+
+`pnpm config list` and `pnpm config get` (without `--json`) imitate an rc file (INI) with best effort.
+They would show rc options in kebab-case, workspace-specific settings in camelCase (since they cannot
+be defined by an rc file).
```

Comment:
Do we really need to change the case? If we moved all non-auth related settings to yaml files, then we can just print the cases as they are.